feat: rewrite todo cards logic

limpbrains opened the pull request:
### Description

Please include a summary of changes and which issues are fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Linked Issues/Tasks

Add any links to GitHub issues or Asana tasks that are relevant to this pull request.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (improving code without creating new functionality)

### Tests

- [ ] Detox test
- [ ] Unit test
- [ ] No test

### Screenshot / Video

Insert relevant screenshot / recording

### QA Notes

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce during the Bitkit testing session. You can also leave a video of the PR in action.

pwltr commented:
Nice, way cleaner than before. Just a couple of findings:

- 'lightningReady' and 'Instant Payments Ready' toast should only show for the first channel, for every extra channel 'transferToSpending' and no toast message
- 'Reset Suggestions' button is not resetting all cards, shows 'lightningReady' card if there is a open channel
- When I disable my PIN code the green card shows again, I think it should stay hidden
- After restore, 'profile' card shows again

pwltr commented on src/store/reselect/todos.ts:
```diff
@@ -1,4 +1,125 @@
+				res.push(transferClosingChannel);
+			} else if (Object.keys(paidOrders.created).length > 0) {
+				res.push(transferToSpendingTodo);
+			} else if (claimableBalance > 0) {
```
I feel like this will falsely trigger in some scenarios when it shouldn't, for example a incoming payment that we haven't claimed yet.
https://docs.rs/lightning/0.0.116/lightning/chain/channelmonitor/enum.Balance.html

limpbrains replied:
Agree will done in #1296

limpbrains commented:
> 'lightningReady' and 'Instant Payments Ready' toast should only show for the first channel, for every extra channel 'transferToSpending' and no toast message

Lets keep it for now. It is still nice to have this confirmatioln when you channel is finally open, even if you already have one

> 'Reset Suggestions' button is not resetting all cards, shows 'lightningReady' card if there is a open channel

fixed

> When I disable my PIN code the green card shows again, I think it should stay hidden

fixed 

> After restore, 'profile' card shows again
will fix in #398 someday :)

pwltr commented:
> Lets keep it for now. It is still nice to have this confirmatioln when you channel is finally open, even if you already have one

Fine with me but maybe check with Aldert as this is not how it is in the design.

> 'Reset Suggestions' button is not resetting all cards, shows 'lightningReady' card if there is a open channel

This issue is still there: 
https://github.com/synonymdev/bitkit/assets/8538369/2062b364-c0f3-4893-8920-d2d2868a8a93

limpbrains commented on src/screens/Settings/Suggestions/index.tsx:
```diff
@@ -7,7 +7,7 @@ import SettingsView from './../SettingsView';
 import Button from '../../../components/Button';
 import { EItemType, IListData } from '../../../components/List';
-import { resetTodos } from '../../../store/actions/todos';
+import { resetHiddelTodos } from '../../../store/actions/todos';
```
thx, fixed

limpbrains commented:
> This issue is still there:

I don't think this is an issue, you have already setup lightning. "Fixing" this requires more complexity.
